### PR TITLE
[THORN-2116] Adding the info about the prebuilt servers

### PR DESCRIPTION
**Pull request by sberyozkin**

### CODE REVIEW

**Comment by sberyozkin** [comment]
See https://wildfly-swarm.gitbooks.io/wildfly-swarm-users-guide/servers.html.
I've only listed two servers, the link to all the available ones is there too.The info about the downloads is only included for the tagged builds.

File: docs/modules/ref_prebuilt-keycloak-server.adoc [hunk 1]
```diff
@@ -0,0 +1,20 @@
+
+[id='prebuilt-keycloak-server_{context}']
+= {Thorntail} Keycloak Server
+
+https://www.keycloak.org/ [Keycloak] is an open source identity and access management solution.
```
Comment (Ladicek) [inline docs/modules/ref_prebuilt-keycloak-server.adoc]: Space between the link and `[Keycloak]` is not OK I guess?

File: docs/modules/ref_prebuilt-keycloak-server.adoc [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+{Thorntail} provides a Keycloak standalone server which can be used to test and run Keycloak Server on top of Thorntail.
+
+ifdef::product[]
```
Comment (Ladicek) [inline docs/modules/ref_prebuilt-keycloak-server.adoc]: `ifndef`?

File: docs/modules/ref_prebuilt-microprofile-server.adoc [hunk 3]
```diff
@@ -0,0 +1,20 @@
+
+[id='prebuilt-microprofile-server_{context}']
+= {Thorntail} Microprofile
+
+https://projects.eclipse.org/projects/technology.microprofile[MicroProfile] is a project aimed at optimizing Enterpise Java for the microservices architecture.
+
+{Thorntail} implements https://projects.eclipse.org/projects/technology.microprofile/releases/microprofile-1.3[MicroProfile 1.3] and provides a hollow MicroProfile standalone server which can be used to deploy any compliant .war applications.
```
Comment (Ladicek) [inline docs/modules/ref_prebuilt-microprofile-server.adoc]: I'd say either "MicroProfile hollow JAR" or "MicroProfile standalone server", but probably not "hollow ... standalone server".

File: docs/modules/ref_prebuilt-microprofile-server.adoc [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+ifdef::product[]
```
Comment (Ladicek) [inline docs/modules/ref_prebuilt-microprofile-server.adoc]: `ifndef`?

File: docs/assemblies/assembly_prebuilt-servers.adoc [hunk 5]
```diff
@@ -0,0 +1,13 @@
+[id='prebuilt-servers_{context}']
+= Pre-built servers
+
+:parent-context: {context}
+
+{Thorntail} has pre-built some https://github.com/thorntail/thorntail/tree/master/standalone-servers[standalone-servers].
+
+include::../modules/ref_prebuilt-microprofile-server.adoc[leveloffset=+1]
+
+include::../modules/ref_prebuilt-keycloak-server.adoc[leveloffset=+1]
```
Comment (Ladicek) [inline docs/assemblies/assembly_prebuilt-servers.adoc]: The Keycloak server is not part of product, is maybe `ifndef` this entire include?

File: docs/modules/ref_prebuilt-keycloak-server.adoc (hunk 1 above)
Comment (sberyozkin) [reply docs/modules/ref_prebuilt-keycloak-server.adoc]: Good catch, it is even visible in the attached image, missed it

File: docs/modules/ref_prebuilt-keycloak-server.adoc (hunk 2 above)
Comment (sberyozkin) [reply docs/modules/ref_prebuilt-keycloak-server.adoc]: That can point to -SNAPSHOT

File: docs/modules/ref_prebuilt-microprofile-server.adoc (hunk 3 above)
Comment (sberyozkin) [reply docs/modules/ref_prebuilt-microprofile-server.adoc]: ok

File: docs/modules/ref_prebuilt-microprofile-server.adoc (hunk 4 above)
Comment (sberyozkin) [reply docs/modules/ref_prebuilt-microprofile-server.adoc]: will point to SNAPSHOT

File: docs/assemblies/assembly_prebuilt-servers.adoc (hunk 5 above)
Comment (sberyozkin) [reply docs/assemblies/assembly_prebuilt-servers.adoc]: What about the tagged upstream release ? In fact the same applies to Download links...
Ladislav, @tradej, does a 'product' affect the tagged upstream releases ?

Comment (Ladicek) [reply docs/assemblies/assembly_prebuilt-servers.adoc]: My understanding is that in the Thorntail community builds, `product` is never set to `true`, that's just for the downstream RHOAR Thorntail product.

Comment (sberyozkin) [reply docs/assemblies/assembly_prebuilt-servers.adoc]: OK, I guess I can indeed exclude a top level link to the Keycloak server if it is 'product' but remove the exclusions from the Downloads (should not be a prob if the links in SNAPSHOT docs won't resolve).

Comment (Ladicek) [reply docs/assemblies/assembly_prebuilt-servers.adoc]: Yea I think so, too.

Comment (tradej) [reply docs/assemblies/assembly_prebuilt-servers.adoc]: In Thorntail community, `product` is not defined. In Fabric8 Launcher builds and on the Portal, it is defined.

**Review by tradej — CHANGES_REQUESTED** [review]
Changes suggested in https://github.com/sberyozkin/thorntail/pull/2

**Comment by michalszynkiewicz** [comment]
It looks to me that all the suggestions are applied and the change looks good to me.
@Ladicek, @tradej you commented  before but didn't approve. Do you think some further modifications should be done on this PR?

**Comment by Ladicek** [comment]
My only objection would be that the list of pre-built servers is quite incomplete. We can add a follow-up task for this, sure.

@tradej Also, "stand-alone"? I've read https://brians.wsu.edu/2016/05/31/standalone-stand-alone/, so  I know stand-alone is correct, but it still feels weird in our context :-)

**Review by Ladicek — APPROVED** [review]

**Comment by sberyozkin** [comment]
Yes, I did not link to all the individual servers but only to those I saw users trying, and because there is a link to the parent directory where the whole list can be seen.

**Comment by sberyozkin** [comment]
Merging is still blocked though, but given that Ladislav has approved now it is all good to be merged ?

**Comment by michalszynkiewicz** [comment]
@sberyozkin it was @tradej's comment not resolved that blocked the merge.

**Comment by Ladicek** [comment]
Filed https://issues.jboss.org/browse/THORN-2216 as a follow-up.

**Pull request merged**